### PR TITLE
Various minor improvements

### PR DESCRIPTION
**Pull request by rusty-key**
Fixing some minor but annoying issues:
— Fix compilation by removing `vendor` from sources
— Add watch script that runs both `bsb` and `tsc` 
— Add formatting script (as formatter in extension doesn't work)
— Add extension run task as suggested by @wokalski
— Add bsb.lock to `gitignore`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,6 @@ client/server
 *~
 _downloads
 .merlin
-lib/
\ No newline at end of file
+lib/
+.bsb.lock
+_esy/
```
Comment (prometheansacrifice) [inline .gitignore]: Did esy run in the vscode plugin folder? It shouldn't have

File: package.json [hunk 2]
```diff
@@ -69,15 +69,18 @@
   "scripts": {
     "clean": "bsb -clean",
     "build": "bsb -make-world",
-    "watch": "tsc -b -w",
+    "watch:bsb": "bsb -make-world -w",
+    "watch:tsc": "tsc -b -w",
```
Comment (prometheansacrifice) [inline package.json]: We dont use tsc anywhere? Did I miss anything?

File: .gitignore (hunk 1 above)
Comment (rusty-key) [reply .gitignore]: maybe it was generated after some experiments. I'll remove it

File: package.json (hunk 2 above)
Comment (rusty-key) [reply package.json]: there is a bunch of TS files in the `client/src`:
https://github.com/arrowresearch/vscode-merlin/tree/master/client/src

Are those not used?

File: .gitignore (hunk 1 above)
Comment (prometheansacrifice) [reply .gitignore]: On second thoughts, let's keep it. Infact, let's drop an esy.json to develop this plugin itself. 

I wanted to make sure, however, that the plugin wasn't creating an esy.json on it's own - it's supposed to create on inside .vscode/esy

File: package.json (hunk 2 above)
Comment (prometheansacrifice) [reply package.json]: No, we can remove them. Apologies, I forgot. Can you please remove them?

**Comment by prometheansacrifice** [comment]
I see that you removed `_esy`. Please ignore my review comment asking to get it back.

PR looks good, nothing blocking. Will merge it once the CI finishes running.

@rusty-key Would you also like to update the lsp in the esy.json to the following in this PR itself?

```json
{
  "dependencies": {
    "@opam/ocaml-lsp-server": "ocaml/ocaml-lsp:ocaml-lsp-server.opam#e5e6ebf9dcf157"
  }
}
```

**Comment by rusty-key** [comment]
@prometheansacrifice, added https://github.com/arrowresearch/vscode-merlin/pull/16/commits/9ddd490e811e48f72544a2c4a463f3b0c31cc4c0

File: package.json (hunk 2 above)
Comment (rusty-key) [reply package.json]: You mean we can remove everything related to TS? I can do it in a separate PR if you want

**Comment by prometheansacrifice** [comment]
I think I miscommunicated - this was useful nevertheless 😄 

I was referring to


https://github.com/arrowresearch/vscode-merlin/blob/9074aaaeaf05c10f2ad3bb36a648c0f9a0a83185/src/Esy.re#L113

**Comment by prometheansacrifice** [comment]
The esy.json you just added to the root of the project should infact look like this


```json
{
  "dependencies": {
    "ocaml": "4.6.x",
    "@esy-ocaml/reason": "*",
    "@opam/ocaml-lsp-server": "ocaml/ocaml-lsp:ocaml-lsp-server.opam#e5e6ebf9dcf157"
  }
}
```

**Comment by prometheansacrifice** [comment]
It needs compiler 4.6 and reason

**Comment by rusty-key** [comment]
@prometheansacrifice, wait. If we are switching to `ocaml-lsp`, we'll need to update binary names everywhere too. Do you mind if I do it in a separate PR so I could properly test it?

**Comment by prometheansacrifice** [comment]
@rusty-key Sure, makes sense! I updated the root esy.json (for us developers) myself

**Pull request merged**